drivers/sensor/pmw3360/pmw3360.c

```diff
@@ -515,6 +518,68 @@ static int update_downshift_time(struct pmw3360_data *dev_data, u8_t reg_addr,
 	return err;
 }
 
+static int update_sample_time(struct pmw3360_data *dev_data,
+			      u8_t reg_addr_lower,
+			      u8_t reg_addr_upper,
+			      u32_t sample_time)
+{
+	/* Set sample time for the Rest1-Rest3 modes.
+	 * Values above 0x09B0 will trigger internal watchdog reset.
+	 */
+	u32_t maxtime = 0x9B0;
+	u32_t mintime = 1;
+
+	if ((sample_time > maxtime) || (sample_time < mintime)) {
+		LOG_WRN("Sample time %u out of range", sample_time);
+		return -EINVAL;
+	}
+
+	LOG_INF("Set sample time to %u ms", sample_time);
+
+	/* The sample time is (reg_value + 1) ms. */
+	sample_time--;
+	u8_t buf[2];
+
+	sys_put_le16((u16_t)sample_time, buf);
+
+	int err = reg_write(dev_data, reg_addr_lower, buf[0]);
+
+	if (!err) {
+		err = reg_write(dev_data, reg_addr_upper, buf[1]);
+	} else {
+		LOG_ERR("Failed to change sample time");
+	}
+
+	return err;
+}
+
+static int toggle_rest_modes(struct pmw3360_data *dev_data, u8_t reg_addr,
+			     bool enable)
+{
+	u8_t value;
+	int err = reg_read(dev_data, reg_addr, &value);
+
+	if (err) {
+		LOG_ERR("Failed to read Config2 register");
+		return err;
+	}
+
+	if (enable) {
+		value |= BIT(PMW3360_REST_EN_POS);
```

Comment:
`WRITE_BIT(value, PMW3360_REST_EN_POS, enable)`

Reply:
if you care

Reply:
done